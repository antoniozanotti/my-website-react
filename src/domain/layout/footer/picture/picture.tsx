import { useTranslations } from "next-intl";
import ImgLookingUpXl from "@images/antonio-zanotti-looking-up-xl.jpg";
import ImgLookingUpLg from "@images/antonio-zanotti-looking-up-lg.jpg";
import ImgLookingUpMd from "@images/antonio-zanotti-looking-up-md.jpg";
import ImgLookingUpSm from "@images/antonio-zanotti-looking-up-sm.jpg";
import ImgLookingUpDefault from "@images/antonio-zanotti-looking-up-default.jpg";

export function Picture() {
  const t = useTranslations("Footer");

  return (
    <picture className="rounded block overflow-hidden mt-4">
      <source media="(min-width: 1280px)" srcSet={ImgLookingUpXl.src} />
      <source media="(min-width: 1024px)" srcSet={ImgLookingUpLg.src} />
      <source media="(min-width: 768px)" srcSet={ImgLookingUpMd.src} />
      <source media="(min-width: 640px)" srcSet={ImgLookingUpSm.src} />
      <img src={ImgLookingUpDefault.src} alt={t("imgAlt")} />
    </picture>
  );
}
