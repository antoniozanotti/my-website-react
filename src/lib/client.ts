import { ApolloClient, HttpLink, InMemoryCache } from "@apollo/client";
import { registerApolloClient } from "@apollo/experimental-nextjs-app-support/rsc";

const TOKEN = process.env.CONTENTFUL_ACCESS_TOKEN;
const SPACE = process.env.CONTENTFUL_SPACE_ID;
const ENVIRONMENT = process.env.CONTENTFUL_ENVIRONMENT_ID;
const URL = `https://graphql.contentful.com/content/v1/spaces/${SPACE}/environments/${ENVIRONMENT}`;

export const { getClient } = registerApolloClient(() => {
  return new ApolloClient({
    cache: new InMemoryCache(),
    link: new HttpLink({
      uri: URL,
      headers: {
        Authorization: `Bearer ${TOKEN}`,
      },
    }),
  });
});
