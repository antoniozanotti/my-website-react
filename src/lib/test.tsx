import React, { ReactElement } from "react";
import {
  render,
  RenderOptions,
  RenderResult,
  Screen,
} from "@testing-library/react";
import { Theme } from "@/domain/theme/theme/theme";
import { Providers } from "../app/providers";
import messages from "@/messages/en.json";

function AllTheProviders({ children }: { children: React.ReactNode }) {
  return (
    <Providers locale="en" messages={messages}>
      <Theme>
        <div data-testid="container">{children}</div>
      </Theme>
    </Providers>
  );
}

function customRender(
  ui: ReactElement,
  options?: Omit<RenderOptions, "wrapper">
): RenderResult {
  return render(ui, { wrapper: AllTheProviders, ...options });
}

function firstComponent(screen: Screen) {
  return screen.getByTestId("container").firstChild;
}

export * from "@testing-library/react";
export * from "@testing-library/user-event";
export { customRender as render, firstComponent };
import "@testing-library/jest-dom";

jest.mock("next/navigation", () => {
  return {
    useRouter() {
      return { push: () => {} };
    },
    useParams: () => ({}),
  };
});
