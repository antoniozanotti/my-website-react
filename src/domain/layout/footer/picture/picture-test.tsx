import { screen, render } from "@/lib/test";
import { Picture } from "./picture";

describe("Picture", () => {
  test("should render picture", async () => {
    const { container } = render(<Picture />);
    const picture = container.querySelector("picture");

    expect(picture).toBeVisible();
  });
});
