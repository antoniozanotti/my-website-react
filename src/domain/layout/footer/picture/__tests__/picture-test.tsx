import { screen, render } from "@/lib/test";
import { Picture } from "../picture";

describe("Picture", () => {
  it("should render picture", async () => {
    const { container } = render(<Picture />);
    const picture = container.querySelector("picture");

    expect(picture).toBeVisible();
    expect(container).toMatchSnapshot();
  });
});
