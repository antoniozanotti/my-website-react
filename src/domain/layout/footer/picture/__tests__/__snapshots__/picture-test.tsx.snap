// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Picture should render picture 1`] = `
<div>
  <div
    class="dark"
  >
    <div
      class="className bg-dark-2 dark:bg-dark-dark-2 text-primary dark:text-dark-primary text-c3 lg:text-c4"
    >
      <div
        data-testid="container"
      >
        <picture
          class="rounded block overflow-hidden mt-4"
        >
          <source
            media="(min-width: 1280px)"
            srcset="/img.jpg"
          />
          <source
            media="(min-width: 1024px)"
            srcset="/img.jpg"
          />
          <source
            media="(min-width: 768px)"
            srcset="/img.jpg"
          />
          <source
            media="(min-width: 640px)"
            srcset="/img.jpg"
          />
          <img
            alt="AntÃ´nio Zanotti looking up"
            src="/img.jpg"
          />
        </picture>
      </div>
    </div>
  </div>
</div>
`;
